File: src/main/java/xyz/upperlevel/hgame/scenario/character/impl/Sfera.java
package xyz.upperlevel.hgame.scenario.character.impl;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import xyz.upperlevel.hgame.scenario.character.Actor;
import xyz.upperlevel.hgame.scenario.character.Character;

public class Sfera implements Character {
    private static final Logger logger = LogManager.getLogger();
    @Override
    public String getName() {
        return "Sfera";
    }

    @Override
    public String getTexturePath() {
        return "prof_sfera.png";
    }

    @Override
    public Actor personify(int id) {
        return new ActorImpl(id, this);
    }

    private class ActorImpl extends Actor {
        private ActorImpl(int id, Character character) {
            super(id, character);
        }

        @Override
        public void move(float offsetX) {
            super.move(offsetX * 0.5f);
        }

        @Override
        public void jump(float strength) {
            super.jump(strength * 0.5f);
        }

        @Override
        public void specialAttack() {
            logger.warn("{}: my special attack!", getName());
        }
    }
}

File: src/main/java/xyz/upperlevel/hgame/world/character/impl/Sfera.kt
package xyz.upperlevel.hgame.world.character.impl

import com.badlogic.gdx.physics.box2d.World
import org.apache.logging.log4j.LogManager
import xyz.upperlevel.hgame.world.character.Entity
import xyz.upperlevel.hgame.world.character.Character
import xyz.upperlevel.hgame.world.character.Player

class Sfera : Character {
    override val name = "Sfera"

    override val texturePath = "prof_sfera.png"

    override fun personify(id: Int, pworld: World): Entity {
        return ActorImpl(id, pworld, this)
    }

    private inner class ActorImpl(id: Int, world: World, character: Character) : Player(id, world, character) {
        override fun jump(strength: Float) {
            super.jump(strength * 0.5f)
        }

        override fun specialAttack() {
            logger.warn("{}: my special attack!", name)
        }
    }

    companion object {
        private val logger = LogManager.getLogger()
    }
}
